app/controllers/api/v1/holidays_controller.rb

```diff
@@ -32,12 +32,7 @@ def available_dates
     api :POST, '/v1/holidays', "Create holiday"
     param_group :holiday
     def create
-      @holiday = Holiday.new(holiday_params)
-      if @holiday.save
-        render json: @holiday, serializer: serializer, status: 201
-      else
-        render json: @holiday.errors.to_json, status: 422
-      end
+      assign_params_and_render_object(Holiday.new, serializer, holiday_params)
```

Comment:
As per CanCanCan comments, no need for Holiday.new here.

CanCanCan is effectively doing the following for you:

    @holiday = Holiday.new
    @holiday.attributes = holiday_params

The source for that is here - https://github.com/CanCanCommunity/cancancan/blob/734c70cb824c58e1efb1473401186abeab023c54/lib/cancan/controller_resource.rb start with `load_resource` method and follow it through to the `assign_attributes` method.